.github/workflows/test.yml

```diff
@@ -66,28 +66,58 @@ jobs:
             flag: --features=gdb
 
   bindeps:
-    name: ${{ matrix.crate }} nightly ${{ matrix.profile.name }}
+    name: ${{ matrix.crate.name }} nightly ${{ matrix.profile.name }}
     runs-on: ubuntu-20.04
     steps:
       - run: sudo apt -o Acquire::Retries=3 update
       - run: sudo apt -o Acquire::Retries=3 install -y musl-tools
       - uses: actions/checkout@v2
       - name: Setup Rust toolchain
         run: rustup show
-      - run: cargo test ${{ matrix.profile.flag }} --target x86_64-unknown-linux-gnu
-        working-directory: src/bin/${{ matrix.crate }}
+      - run: cargo test ${{ matrix.profile.flag }} --target x86_64-unknown-linux-gnu --manifest-path ${{ matrix.crate.path }}/Cargo.toml
     strategy:
       fail-fast: false
       matrix:
         crate:
-          - shim-kvm
-          - shim-sgx
-          - exec-wasmtime
+          - { name: exec-wasmtime, path: src/bin/exec-wasmtime }
+          - { name: shim-kvm, path: src/bin/shim-kvm }
+          - { name: shim-sgx, path: src/bin/shim-sgx }
+          - { name: sallyport, path: crates/sallyport }
         profile:
           - name: debug
           - name: release
             flag: --release
 
+  miri:
+    name: ${{ matrix.crate.name }} miri ${{ matrix.profile.name }}
+    runs-on: ubuntu-20.04
+    steps:
+      - run: sudo apt -o Acquire::Retries=3 update
+      - run: sudo apt -o Acquire::Retries=3 install -y musl-tools
+      - uses: actions/checkout@v2
+      - name: Setup Rust toolchain
+        run: rustup show
+      - uses: actions-rs/cargo@v1
+        name: cargo miri setup
+        with:
+          command: miri
+          args: setup --manifest-path ${{ matrix.crate.path }}/Cargo.toml
+      - uses: actions-rs/cargo@v1
+        name: cargo miri test
+        env:
+          MIRIFLAGS: "-Zmiri-disable-isolation"
+        with:
+          command: miri
+          args: test --manifest-path ${{ matrix.crate.path }}/Cargo.toml
+    strategy:
+      fail-fast: false
+      matrix:
+        crate:
+          - { name: sallyport, path: crates/sallyport }
```

Comment:
We probably want to enable miri for other crates as well (eventually?)

Reply:
exactly.. but not now